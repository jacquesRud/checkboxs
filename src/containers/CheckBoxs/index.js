import React, { useState } from "react";
import { sizeArrayCheckbox } from "../../Constants";
import CheckBox from "../../components/CheckBox";

export default function CheckBoxs() {
  const [checkBoxs, setCheckBoxs] = useState(
    new Array(sizeArrayCheckbox).fill(false)
  );

  return (
    <>
      <div style={{ display: "flex" }}>
        <CheckBox
          enabled={checkBoxs.every((checkBox) => checkBox)}
          onChange={(newVal) => {
            updateAll(newVal);
          }}
        />{" "}
        Select All
      </div>
      {checkBoxs.map((checkBox, index) => (
        <div style={{ display: "flex" }} id={index}>
          <CheckBox
            enabled={checkBox}
            onChange={(newVal) => {
              onChange(newVal, index);
            }}
          />
          Item {index}
        </div>
      ))}
    </>
  );

  function onChange(newVal, index) {
    checkBoxs[index] = newVal;
    setCheckBoxs([...checkBoxs]);
  }

  function updateAll(newVal) {
    setCheckBoxs(new Array(sizeArrayCheckbox).fill(newVal));
    console.log(checkBoxs);
  }
}
