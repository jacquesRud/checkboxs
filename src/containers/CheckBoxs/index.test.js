import React from "react";
import { render } from "@testing-library/react";
import CheckBoxs from "./";

describe("Checkboxs", () => {
  test("Render", () => {
    const { container } = render(<CheckBoxs />);
    expect(container).toMatchSnapshot();
  });
});
