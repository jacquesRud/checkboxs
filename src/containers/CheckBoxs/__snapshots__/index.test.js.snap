// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkboxs Render 1`] = `
<div>
  <div
    style="display: flex;"
  >
    <div
      style="cursor: pointer;"
    >
      [ 
       
       ]
    </div>
     
    Select All
  </div>
  <div
    id="0"
    style="display: flex;"
  >
    <div
      style="cursor: pointer;"
    >
      [ 
       
       ]
    </div>
    Item 
    0
  </div>
  <div
    id="1"
    style="display: flex;"
  >
    <div
      style="cursor: pointer;"
    >
      [ 
       
       ]
    </div>
    Item 
    1
  </div>
  <div
    id="2"
    style="display: flex;"
  >
    <div
      style="cursor: pointer;"
    >
      [ 
       
       ]
    </div>
    Item 
    2
  </div>
  <div
    id="3"
    style="display: flex;"
  >
    <div
      style="cursor: pointer;"
    >
      [ 
       
       ]
    </div>
    Item 
    3
  </div>
  <div
    id="4"
    style="display: flex;"
  >
    <div
      style="cursor: pointer;"
    >
      [ 
       
       ]
    </div>
    Item 
    4
  </div>
</div>
`;
